### PR TITLE
Feature/inventory

### PR DESCRIPTION
**Pull request by Yanis-Kouidri**
Creation of inventory with model view and controller

### CODE REVIEW

File: src/gameengine/inventory/controller/InventoryKeyController.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package gameengine.inventory.controller;
+
```
Comment (CedricAbd) [inline src/gameengine/inventory/controller/InventoryKeyController.java:2]: This class should have comments and javadoc

File: src/gameengine/inventory/model/InventoryFullException.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package gameengine.inventory.model;
+
```
Comment (CedricAbd) [inline src/gameengine/inventory/model/InventoryFullException.java:2]: Add purpose author and version

File: src/gameengine/inventory/model/Item.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
+package gameengine.inventory.model;
+
```
Comment (CedricAbd) [inline src/gameengine/inventory/model/Item.java:2]: Add purpose author and version

File: src/gameengine/inventory/model/NotEmptyPlaceException.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package gameengine.inventory.model;
+
```
Comment (CedricAbd) [inline src/gameengine/inventory/model/NotEmptyPlaceException.java:2]: Add purpose author and version

File: src/gameengine/inventory/model/NothingToRemoveException.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package gameengine.inventory.model;
+
```
Comment (CedricAbd) [inline src/gameengine/inventory/model/NothingToRemoveException.java:2]: Add purpose author and version

File: src/gameengine/inventory/view/InventoryBar.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
+package gameengine.inventory.view;
+
+import gameengine.inventory.model.Inventory;
+
+import java.awt.*;
+
```
Comment (CedricAbd) [inline src/gameengine/inventory/view/InventoryBar.java:6]: Add purpose author and version

File: src/gameengine/inventory/view/InventoryMenu.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
+package gameengine.inventory.view;
+
+import gameengine.inventory.model.Inventory;
+
+import java.awt.*;
+
```
Comment (CedricAbd) [inline src/gameengine/inventory/view/InventoryMenu.java:6]: Add purpose author and version

File: src/gameengine/inventory/view/InventoryPanel.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
+package gameengine.inventory.view;
+
+import gameengine.inventory.model.Inventory;
+import gameengine.inventory.model.Item;
+
+import javax.swing.*;
+import javax.swing.border.Border;
+import java.awt.*;
+
```
Comment (CedricAbd) [inline src/gameengine/inventory/view/InventoryPanel.java:9]: Add purpose author and version

File: test/gameengine/inventory/model/InventoryTest.java [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.swing.*;
+
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.*;
+
```
Comment (CedricAbd) [inline test/gameengine/inventory/model/InventoryTest.java:13]: Add purpose author and version

File: test/gameengine/inventory/model/ItemTest.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
+package gameengine.inventory.model;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.swing.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+
```
Comment (CedricAbd) [inline test/gameengine/inventory/model/ItemTest.java:10]: Add purpose author and version

File: test/gameengine/inventory/view/InventoryBarViewTest.java [hunk 11]
```diff
@@ -0,0 +1,55 @@
+package gameengine.inventory.view;
+
+import gameengine.inventory.model.Inventory;
+import gameengine.inventory.model.Item;
+import gameengine.inventory.model.NotEmptyPlaceException;
+
+import javax.swing.*;
+import java.awt.*;
+
```
Comment (CedricAbd) [inline test/gameengine/inventory/view/InventoryBarViewTest.java:9]: Add purpose author and version

File: test/gameengine/inventory/view/InventoryMenuViewTest.java [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+
+import gameengine.inventory.controller.InventoryKeyController;
+import gameengine.inventory.model.Inventory;
+import gameengine.inventory.model.Item;
+import gameengine.inventory.model.NotEmptyPlaceException;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.Random;
+
```
Comment (CedricAbd) [inline test/gameengine/inventory/view/InventoryMenuViewTest.java:11]: Add purpose author and version

**Review by CedricAbd — APPROVED** [review]
Approved. Just some modifications to do on the comments.

**Comment by Yanis-Kouidri** [comment]
Inventory : OK

**Pull request merged**